pkg/net/host.go

```diff
@@ -143,3 +154,15 @@ func ParseHost(s string) (*Host, error) {
 		IsPortSet: isPortSet,
 	}, nil
 }
+
+// IPv6 can be embedded with square brackets.
+func trimIPv6(host string) (string, error) {
+	// `missing '[' in host` error is already handled in `SplitHostPort`
+	if host[len(host)-1] == ']' {
+		if host[0] != '[' {
```

Comment:
Isn't this a redundant check, if it is already covered in `SplitHostPort`

Reply:
The comment was misleading. It should have been `missing ']' in host` error is already handled in SplitHostPort`. Changed it.